compiler/src/dmd/statementsem.d

```diff
@@ -558,6 +558,24 @@ Statement statementSemanticVisit(Statement s, Scope* sc)
     {
         /* https://dlang.org/spec/statement.html#do-statement
          */
+
+        /* Rewrite: do { body } while (cond);
+         * To:      for (; true; } { body; if (!cond) break; }
+         */
+        Expression notCond = new NotExp(ds.condition.loc, ds.condition);
+        Statement ifNotCond = new IfStatement(ds.loc, null, notCond, ds._body, new BreakStatement(ds.loc, null), ds.endloc);
+
+        Statement pair = new CompoundStatement(ds._body.loc, ds._body, ifNotCond);
+        Statement scopePair = new ScopeStatement(pair.loc, pair, pair.loc);
+
+        Expression _true = new IntegerExp(ds.loc, 1, Type.tbool);
```

Comment:
use a cached literal, `IntegerExp.literal!1` (or whatever it is called)

Reply:
good idea